exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -281,11 +282,25 @@ object SchemaUtils {
      *
      * @param databases the names of the databases
      * @param inBatch flag to perform database creation in a single batch
+     *
+     * For PostgreSQL, calls to this function should be preceded by connection.autoCommit = true,
+     * and followed by connection.autoCommit = false.
+     * @see org.jetbrains.exposed.sql.tests.shared.ddl.CreateDatabaseTest
      */
     fun createDatabase(vararg databases: String, inBatch: Boolean = false) {
-        with(TransactionManager.current()) {
-            val createStatements = databases.flatMap { listOf(currentDialect.createDatabase(it)) }
-            execStatements(inBatch, createStatements)
+        val transaction = TransactionManager.current()
+        try {
+            with(transaction) {
+                val createStatements = databases.flatMap { listOf(currentDialect.createDatabase(it)) }
```

Comment:
Can you please also fix it to `databases.map { currentDialect.createDatabase(it) }` here and below in `dropDatabase`?

Reply:
@Tapac Why? We are currently mapping every `String` to a list, to end up with a list of lists, which we then flatten. Changing it to `map` would cause a compilation error, unless we wrap `execStatements(inBatch, createStatements)` with a `forEach`.